### ISSUE TITLE
Analysis checkers should catch updates declared as queries and vice versa

### ISSUE THREAD

**Issue by nigredo-tori** [tpolecat/doobie#673]
I made a silly mistake yesterday, defining something like this:
```scala
sql"DELETE FROM foo RETURNING lo_unlink(bar)".update
```
Basically, I used `RETURNING` clause to run an effect for each row being deleted, and declared this as an `Update0`, intending to ignore the results.

The problem is, doobie analysis checkers didn't catch this, resulting in a runtime exception when executing the `Update0`:
```
org.postgresql.util.PSQLException: A result was returned when none was expected.
```
I'd like for the checkers to properly diagnose this problem in the future.

**Comment by tpolecat** [comment]
Good idea. The difference between `update` and `query` is basically whether you expect a resultset or not. We can certanly check this.
